docs/developers/tutorials/staking-contract.md

```diff
@@ -0,0 +1,635 @@
+---
+id: staking-contract
+title: Staking smart contract tutorial
+---
+
+# Introduction
+
+This tutorial aims to teach you how to write a simple staking contract, and to illustrate and correct the common pitfalls new smart contract developers might fall into.  
+
+If you find anything not answered here, feel free to ask further questions on the Elrond Developers Telegram channel: https://t.me/ElrondDevelopers
+
+# Prerequisites
+
+## erdpy
+
+First and foremost, you need to have erdpy installed: https://docs.elrond.com/sdk-and-tools/erdpy/installing-erdpy/
+
+If you already have erdpy installed, make sure to update it to the latest version, using the same instructions as for the installation.
+
+We're going to use erdpy for interacting with our contracts, so if you need more details about some of the steps we will perform, you can check here for more detailed explanations regarding what each command does: https://docs.elrond.com/sdk-and-tools/erdpy/smart-contract-interactions/
+
+## Rust
+
+Once you have erdpy installed, you also have to install Rust through it, and the VM tools for testing:
+```
+erdpy deps install rust
+
+erdpy deps install vmtools --overwrite
+```
+
+If you installed Rust already without erdpy, you might run into some issues when building your smart contracts. It's recommended to uninstall Rust, and install it through erdpy instead.
+
+Example of error:
+```
+error[E0554]: #![feature] may not be used on the stable release channel
+--> /home/user/elrondsdk/vendor-rust/registry/src/github.com-1ecc6299db9ec823/elrond-wasm-derive-0.33.0/src/lib.rs:4:12
+```
+
+## VSCode and rust-analyser extension
+
+VSCode: https://code.visualstudio.com/
+
+Assuming you're on Ubuntu, download the `.deb` version. Go to that folder:
+- open folder in terminal
+- run the following command: `sudo dpkg -i downloaded_file_name`
+
+rust-analyser: https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer
+Elrond VSCode extension: https://marketplace.visualstudio.com/items?itemName=Elrond.vscode-elrond-ide
+
+Both can be easily installed from the "Extensions" menu in VSCode.  
+
+# Creating the contract
+
+Run the following command in the folder in which you want your smart contract to be created:
+```
+erdpy contract new staking-contract --template empty
+```
+
+Open the generated folder, and you should have the following structure:  
+![img](/developers/staking-contract-tutorial-img/folder_structure.png)
+
+For now, comment all the code in the `empty_rust_test.rs` file (ctrl + "A", then ctrl + "/"). Otherwise, it will keep popping up errors as we modify the contract's code.
```

Comment:
Specify the exact location of the file:
_in the `./tests/empty_rust_test.rs` file._